File: src/Python/255.VerifyPreorderSequenceInBinarySearchTree.py

class Solution:
    def verifyPreorde(self, nums):
        minnum = -99999999
        stack = []
        for num in nums:
            if num < minnum:
                return False
            while len(stack) > 0 and num > stack[-1]:
                minnum = stack.pop()
            stack.append(num)
        return True

    
if __name__ == "__main__":
    solution = Solution()

File: src/CSharp/255.VerifyPreorderSequenceInBinarySearchTree.cs

using System;
using System.Linq;
using System.Text;
using System.Collections.Generic;

namespace PeefyLeetCode.VerifyPreorderSequenceInBinarySearchTree
{
    public class Solution
    {
        public bool VerifyPreorde(int [] nums)
        {
            var min = int.MinValue;
            var stack = new Stack<int>();
            foreach (var num in nums)
            {
                if (num < min)
                    return false;
                while (stack.Count > 0 && num > stack.Peek())
                {
                    min = stack.Pop();
                }
                stack.Push(num);
            }
            return true;
        }
    }
}